### PR TITLE
Document to ignore drift detection

### PR DESCRIPTION
**Pull request by funera1**
**What this PR does / why we need it**:
Document to ignore drift detection the specified fields.

I don't guess that the `Syntax` section in the `configuration-drift-detection.md` gives a very good explanation, so any help with a better explanation would be appreciated. Thank you.

**Which issue(s) this PR fixes**:

Fixes part of #4190

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 1]
```diff
@@ -666,3 +667,9 @@ A wrapper of type `int` to represent percentage data. Basically, you can pass `1
 |-|-|-|-|
 | commitMessage | string | The commit message used to push after replacing values. Default message is used if not given. | No |
 | replacements | [][EventWatcherReplacement](#eventwatcherreplacement) | List of places where will be replaced when the new event matches. | Yes |
+
+### DriftDetection
+
+| Field | Type | Description | Required |
+|-|-|-|-|
+| ignoreFields | []string |  Application configuration fields to ignore drift detection. | No |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:675]:
```suggestion
| ignoreFields | []string |  List of fields path in manifests, which its diff should be ignored. | No |
```

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ spec:
 | postSync | [PostSync](#postsync) | Additional configuration used as extra actions once the deployment is triggered. | No |
 | variantLabel | [KubernetesVariantLabel](#kubernetesvariantlabel) | The label will be configured to variant manifests used to distinguish them. | No |
 | eventWatcher | [][EventWatcher](#eventwatcher) | List of configurations for event watcher. | No |
+| driftDetection | [DriftDetection](#driftdetection) | Additional configuration for drift detection between application’s configuration of live state and application’s configuration on git. | No |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:40]:
```suggestion
| driftDetection | [DriftDetection](#driftdetection) | Configuration for drift detection. | No |
```

File: docs/content/en/docs-dev/user-guide/configuration-reference.md (hunk 1 above)
Comment (funera1) [reply docs/content/en/docs-dev/user-guide/configuration-reference.md:675]: Thank you:+1:

File: docs/content/en/docs-dev/user-guide/configuration-reference.md (hunk 2 above)
Comment (funera1) [reply docs/content/en/docs-dev/user-guide/configuration-reference.md:40]: Thank you :+1:

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 3]
```diff
@@ -666,3 +667,9 @@ A wrapper of type `int` to represent percentage data. Basically, you can pass `1
 |-|-|-|-|
 | commitMessage | string | The commit message used to push after replacing values. Default message is used if not given. | No |
 | replacements | [][EventWatcherReplacement](#eventwatcherreplacement) | List of places where will be replaced when the new event matches. | Yes |
+
+### DriftDetection
+
+| Field | Type | Description | Required |
+|-|-|-|-|
+| ignoreFields | []string |  List of fields path in manifests, which its diff should be ignored. | No |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:675]:
```suggestion
| ignoreFields | []string | List of fields path in manifests, which its diff should be ignored. | No |
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 4]
```diff
@@ -53,3 +53,54 @@ This status means the application is deploying and the configuration drift detec
 This feature is automatically enabled for all applications.
 
 You can change the checking interval as well as [configure the notification](../../managing-piped/configuring-notifications/) for these events in `piped` configuration.
+
+### Enable to ignore drift detection for specific fields
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:57]:
```suggestion
### Ignore drift detection for specific fields
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 5]
```diff
@@ -53,3 +53,54 @@ This status means the application is deploying and the configuration drift detec
...
+
+>  Note: This feature is currently supported for only kubernetes application.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:59]:
```suggestion
>  Note: This feature is currently supported for only Kubernetes application.  
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 6]
```diff
@@ -53,3 +53,54 @@ This status means the application is deploying and the configuration drift detec
 This feature is automatically enabled for all applications.
 
 You can change the checking interval as well as [configure the notification](../../managing-piped/configuring-notifications/) for these events in `piped` configuration.
+
+### Enable to ignore drift detection for specific fields
+
+>  Note: This feature is currently supported for only kubernetes application.  
+
+This feature ignores drift detection for the specified fields. In other words, even if the specified fields are different between live state and Git, the application status will not be `Out of Sync`.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:61]:
```suggestion
You can also ignore drift detection for specified fields in your application manifests. In other words, even if the selected fields have different values between live state and Git, the application status will not be set to `Out of Sync`.
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 7]
```diff
@@ -53,3 +53,54 @@ This status means the application is deploying and the configuration drift detec
...
+#### Syntax
+There are two kinds of block collection in yaml.
+
+Sequence: This block collection is indicated each entry with a dash and space ("`- `").  
+Mapping: This block collection is indicated each entry with a colon and space("`: `").
+
+Mapping indexes are specified, for example `.foo`.  
+Sequence indexes are specified, for example `.2`.  
+However, the first index doesn't need the `.`.
+
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:74]: I think we don't need this blog, users are expected to understand `yaml` syntax or if not we have to add this to every other places 👀

Comment (funera1) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:74]: I see. I will remove this section. 
Thank you:smile:

Comment (khanhtc1202) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md]: Also, should move L63 under the example as well

Comment (funera1) [reply docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md]: I fixed it 😃

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 8]
```diff
@@ -53,3 +53,44 @@ This status means the application is deploying and the configuration drift detec
...
+### Ignore drift detection for specific fields
+
+>  Note: This feature is currently supported for only Kubernetes application.  
+
+You can also ignore drift detection for specified fields in your application manifests. In other words, even if the selected fields have different values between live state and Git, the application status will not be set to `Out of Sync`.
+
+To use this feature, set parameters in the application configuration. For more information, see the [configuration reference](../../configuration-reference/#driftdetection) and the following `Example` section.
+
+#### Example
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:65]:
```suggestion
For example, suppose you have the application's manifest as below 
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 9]
```diff
@@ -53,3 +53,44 @@ This status means the application is deploying and the configuration drift detec
...
 
 You can change the checking interval as well as [configure the notification](../../managing-piped/configuring-notifications/) for these events in `piped` configuration.
+
+### Ignore drift detection for specific fields
+
+>  Note: This feature is currently supported for only Kubernetes application.  
+
+You can also ignore drift detection for specified fields in your application manifests. In other words, even if the selected fields have different values between live state and Git, the application status will not be set to `Out of Sync`.
+
+To use this feature, set parameters in the application configuration. For more information, see the [configuration reference](../../configuration-reference/#driftdetection) and the following `Example` section.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:63]:
```suggestion
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 10]
```diff
@@ -53,3 +53,44 @@ This status means the application is deploying and the configuration drift detec
...
+            - hello
+          image: gcr.io/pipecd/helloworld:v1.0.0
+          name: helloworld
+```
+
+If you want to ignore the drift detection for the two sceans
+- pod's replicas
+- `helloworld` container's args
+
+you may add the follow statements to `app.pipe.yaml`.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:87]:
```suggestion
Add the following statements to `app.pipe.yaml` to ignore diff on those fields.
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 11]
```diff
@@ -53,3 +53,44 @@ This status means the application is deploying and the configuration drift detec
...
+```yaml
+spec:
+  ...
+  driftDetection:
+    ignoreFields:
+      - spec.replicas
+      - spec.template.spec.containers.0.args
+```
+
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:96]:
```suggestion
For more information, see the [configuration reference](../../configuration-reference/#driftdetection).
```

File: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md [hunk 12]
```diff
@@ -53,3 +53,44 @@ This status means the application is deploying and the configuration drift detec
...
+            - hello
+          image: gcr.io/pipecd/helloworld:v1.0.0
+          name: helloworld
+```
+
+If you want to ignore the drift detection for the two sceans
+- pod's replicas
+- `helloworld` container's args
+
+Add the following statements to `app.pipe.yaml` to ignore diff on those fields.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md:86]:
```suggestion
Add the following statements to `app.pipecd.yaml` to ignore diff on those fields.
```

**Review by khanhtc1202 — APPROVED** [review]
Nice, thank you 🙌

**Pull request merged**